File: java/lib/ProtocGenJava/src/main/java/com/tonicdesign/protocgenjava/Reader.java
package com.tonicdesign.protocgenjava;

import java.util.Map;

public interface Reader {
    int readTag();
    byte readByte();
    long readVarInt();
    long readVarUInt();
    boolean readBool();
    int readUInt32();
    long readUInt64();
    float readFloat32();
    double readFloat64();
    String readString();
    int pushLimit(int limit);
    void popLimit(int limit);
    void pushTagMap(Map<String, TagMapValue> map);
    void popTagMap();
}

File: swift/lib/ProtocGenSwift/ProtocGenSwift/Reader.swift
import Foundation

public protocol Reader {
    static func fromBuffer(data: NSData) -> Reader?
    func readTag() -> Int
    func readByte() -> UInt8
    func readVarInt() -> Int
    func readVarUInt() -> UInt
    func readBool() -> Bool
    func readFloat32() -> Float32
    func readFloat64() -> Float64
    func readUInt32() -> UInt32
    func readUInt64() -> UInt64
    func readString() -> String
    func pushLimit(limit: Int) -> Int
    func popLimit(limit: Int)
    func pushTagMap(map: [String:(Int, Bool)])
    func popTagMap()
}